worker/src/vectorize.rs

```diff
@@ -0,0 +1,319 @@
+use std::collections::HashMap;
+
+use crate::{send::SendFuture, EnvBinding, Result};
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::{JsCast, JsValue};
+use wasm_bindgen_futures::JsFuture;
+use worker_sys::types::VectorizeIndex as VectorizeIndexSys;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+/// Supported distance metrics for an index.
+/// Distance metrics determine how other "similar" vectors are determined.
+pub enum VectorizeDistanceMetric {
+    Euclidean,
+    Cosine,
+    DotProduct,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+/// Information about the configuration of an index.
+pub enum VectorizeIndexConfig {
+    Preset {
+        preset: String,
+    },
+    Custom {
+        dimensions: u16,
+        metric: VectorizeDistanceMetric,
+    },
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+/// Metadata about an existing index.
+pub struct VectorizeIndexDetails {
```

Comment:
missing [vectorsCount](https://workers-types.pages.dev/#VectorizeIndexDetails.vectorsCount)

Reply:
Thanks for catching this. With Vectorize V2, the `describe` operation should return the `VectorizeIndexInfo` rather than `VectorizeIndexDetails` type: https://workers-types.pages.dev/#Vectorize.describe. I have fixed the types.